app/controllers/interactions_controller.rb

```diff
@@ -0,0 +1,6 @@
+class InteractionsController < ApplicationController
+  def index
+    @service = Service.find_by_slug!(params[:service_slug])
+    @interactions = Service.find(@service.id).interactions
```

Comment:
This can be `@interactions = @service.interactions` - we've already got the `Service` instance (we even use it in this 2nd `Service.find`) so there's no need to look it up twice.